### ISSUE TITLE
Global Copy, Cut, Paste and Duplicate Elements

### ISSUE THREAD

**Issue by jauyong** [GoogleForCreators/web-stories-wp#519]
## Feature Description

As an editor I want to be able to use the copy, cut, paste and duplicate functions 

Note: Pasting text will be handled in #760 

[Feature Brief
](https://docs.google.com/document/d/1_3-_v4zH6bW9hOplOO8wYxj8l4C0soMgEbD7XLiXrrg/edit#heading=h.y61q0s903u48)

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria

1. Make copy, cut, duplicate and paste work globally. 
2. copy, cut, duplicate and paste should work globally everywhere no matter what is focused. Except when a text input is in focus.
3. For duplication and paste, the newly created element becomes focused/selected right after.
3. When multiple elements are selected, the function (Copy, Cut, Paste and Duplicate) treats all selected items as one
4. New elements that are pasted or duplicated should have their own unique ID
4. Keyboard shortcuts should work:
5. ⌘  + C to copy an element in the clipboard.
6. ⌘  + X to move (delete from current page and copy) an element into the clipboard.
7. ⌘  + D to duplicate the current element.
8. ⌘  + V to paste either:
   10a. A previously copied element into the current page (at cursor position)
   10b. A previously copied text content, either
  - Into the currently selected element, if in crop mode, or:
  - As new text element
   10c. A previously copied image or video (which then gets uploaded, see Uploading Media)

Note: These keyboard shortcuts work globally – the workspace does not have to have focus. The only exception are situations when the user is expecting to edit text:
1. Text Inputs: In this case, we fallback to browser native cut/copy/paste. See #760 
2. Color picker In this case, we assume the author wants to cut/copy/paste the hex value)

--

**Comment by dvoytenko** [comment]
This should be renamed to "Global paste for file upload" or such. The element-based copy/cut is already done.

**Comment by spacedmonkey** [comment]
> This should be renamed to "Global paste for file upload" or such. The element-based copy/cut is already done.

I don't think it should. This ticket should handle all copy/cut/paste globally. As these things are so heavily linked.

**Comment by pbakaus** [comment]
Wrote it out here: https://docs.google.com/document/d/1_3-_v4zH6bW9hOplOO8wYxj8l4C0soMgEbD7XLiXrrg/edit#heading=h.y61q0s903u48

@spacedmonkey let me know if you can think of more edge cases, other than color picker and text inputs.
